ImageSegmentation/ImageSegmentation/TorchModule.mm

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2020 Facebook, Inc. and its affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+
+
+#import "TorchModule.h"
+#import <LibTorch/LibTorch.h>
+#import <CoreImage/CoreImage.h>
+#import <ImageIO/ImageIO.h>
+#import <UIKit/UIKit.h>
+
+@implementation TorchModule {
+ @protected
+  torch::jit::script::Module _impl;
+}
+
+UIImageView* m_imageView;
+
+- (nullable instancetype)initWithFileAtPath:(NSString*)filePath {
+  self = [super init];
+  if (self) {
+    try {
+        
+      _impl = torch::jit::load(filePath.UTF8String);
+      _impl.eval();
+    } catch (const std::exception& exception) {
+      NSLog(@"%s", exception.what());
+      return nil;
+    }
+  }
+  return self;
+}
+
+- (void)setImageView:(UIImageView*)imageView {
```

Comment:
Should be moved to the UI layer which will be guaranteed to be called via Main thread.

Reply:
All great points and will change. Thanks! About 3 - the current NLP demo uses a kind of mysterious model reddit.pt as we don't easily know how the model is built and what input and output should be expected. The NMT demo app is built with 3 different goals: as the first example for developers who try to use a model with source Python code for training; to show how to convert the original Python inference code to iOS with correct inputs and outputs; to show a more complicated case of using multiple models (encoder and decoder) in an iOS app (also with a tuple of tensors as output). Also, it may be desirable to pair each additional CV model based mobile app with an NLP model based one - there's a good balance of CV and NLP demo apps at TF Lite https://www.tensorflow.org/lite/examples.